tests/suites/model/destroy.sh

```diff
@@ -0,0 +1,46 @@
+# Tests if Juju tracks the model properly through deletion.
+#
+# In normal behavior Juju should drop the current model selection if that
+# model is destroyed. This will fail if Juju does not drop it's current
+# selection.
+run_model_destroy() {
+	# Echo out to ensure nice output to the test suite.
+	echo
+
+	# The following ensures that a bootstrap juju exists.
+	file="${TEST_DIR}/test-model-destroy.log"
+	ensure "model-destroy" "${file}"
+
+	echo "Ensure current model is 'model-destroy'"
+	juju models --format json | jq -r '."current-model"' | check model-destroy
+
+	echo "Add new model 'model-new'"
+	juju add-model model-new
+
+	echo "Ensure current model is 'model-new'"
+	juju models --format json | jq -r '."current-model"' | check model-new
+
+	echo "Destroy model 'model-new'"
+	destroy_model "model-new"
+	juju switch :model-destroy
+
+	echo "Ensure current model is 'model-destroy'"
+	juju models --format json | jq -r '."current-model"' | check model-destroy
+
+	destroy_model "model-destroy"
+}
```

Comment:
One small bit of coverage missing
https://github.com/juju/juju/blob/ac9ddcae708eca8ec2fd885a14fb27a291e424da/acceptancetests/assess_destroy_model.py#L68-L70

The python based test ensures that the created model can't be found

Reply:
```sh
destroy_model "model-new"
is_destroyed=$(juju models --format json | jq -r '.models[] | select(."short-name" == "model-new")')
if [[ -z ${is_destroyed} ]]; then is_destroyed=true; fi
check_contains "${is_destroyed}" true
```
Added explicit model deletion check.